iptvtools/models.py

```diff
@@ -144,20 +144,23 @@ def filter(self):
         random.shuffle(urls)
         pbar = tqdm(urls, ascii=True)
         for url in pbar:
-            time.sleep(self.args.interval)
             status = 'OK'
-            if self.args.min_height or self.args.resolution_on_title:
-                height = utils.check_stream(url, self.args.timeout)
-                if height == 0:
+            if not self.args.no_connectivity_check:
+                time.sleep(self.args.interval)
+                if self.args.min_height or self.args.resolution_on_title:
```

Comment:
I would prefer to follow the `reduce nesting`[1] rule. Here we just have three branches:

1. No connection check, mark as skipped.
2. Need height/resolution info, check stream.
3. Normal connection check, check_connectivity.

[1] https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html